IdentityCore/src/MSIDBrokerFlightProvider.m

```diff
@@ -0,0 +1,110 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+#import "MSIDBrokerFlightProvider.h"
+#import "NSJSONSerialization+MSIDExtensions.h"
+
+@interface MSIDBrokerFlightProvider()
+
+@property (nonatomic, nullable) NSDictionary *clientFlightsPayload;
+
+@end
+
+@implementation MSIDBrokerFlightProvider
+
+- (instancetype)initWithBase64EncodedFlightsPayload:(nullable NSString *)base64EncodedFlightsPayload
+{
+    self = [super init];
+    
+    if (self)
+    {
+        if ([NSString msidIsStringNilOrBlank:base64EncodedFlightsPayload])
+        {
+            MSID_LOG_WITH_CTX(MSIDLogLevelInfo,nil, @"Broker client flights is nil or empty");
+            return nil;
+        }
+        
+        NSDictionary *clientFlightsDict = nil;
+        
+        NSData *decodedJsonData =  [[base64EncodedFlightsPayload msidBase64UrlDecode] dataUsingEncoding:NSUTF8StringEncoding];
+        if (decodedJsonData && [decodedJsonData length])
+        {
+            clientFlightsDict = [NSJSONSerialization msidNormalizedDictionaryFromJsonData:decodedJsonData error:nil];
+            
+            if (![clientFlightsDict isKindOfClass:[NSDictionary class]])
+            {
+                MSID_LOG_WITH_CTX(MSIDLogLevelWarning,nil, @"Invalid broker client flight format");
+                return nil;
+            }
+        }
+        else
+        {
+            MSID_LOG_WITH_CTX(MSIDLogLevelWarning,nil, @"Failed to decode base64encoded client flights from broker");
+            return nil;
+        }
+        
+        if (clientFlightsDict)
+        {
+            self.clientFlightsPayload = clientFlightsDict;
+            MSID_LOG_WITH_CTX(MSIDLogLevelInfo, nil, @"Client flights from broker is decoded successfully");
+        }
+    }
+    
+    return self;
+}
+
+#pragma mark - MSIDFlightManagerInterface
+
+- (BOOL)boolForKey:(nonnull NSString *)flightKey
+{
+    if (self.clientFlightsPayload)
+    {
+        id value = self.clientFlightsPayload[flightKey];
+        
+        if ([value isKindOfClass:[NSNumber class]] || [value isKindOfClass:[NSString class]])
+        {
+            return [value boolValue];
+        }
+    }
+    
+    return NO;
+}
+
+- (nullable NSArray<NSString *> *)stringsForFlightKey:(nonnull NSString *)flightKey
+{
+    if (self.clientFlightsPayload)
+    {
+        id value = self.clientFlightsPayload[flightKey];
+        
+        if ([value isKindOfClass:[NSArray class]])
+        {
+            return value;
```

Comment:
nit: what if the elemet in the array is not `NSString`

Reply:
It was returning `nil`. I've now updated this method to return a single NSString value from the flight; in case the value from ECS is not a string, this method is also returning `nil`.